recipes-bsp/imx-sc-firmware/imx-sc-firmware_%.bbappend

```diff
@@ -4,24 +4,22 @@ SC_FIRMWARE_NAME:imx8qxpb0-var-som = "mx8qx-var-som-scfw-tcm.bin"
 SC_FIRMWARE_NAME:imx8qm-var-som = "mx8qm-var-som-scfw-tcm.bin"
 
 SC_MX_FAMILY ?= "INVALID"
-SC_MX8_FAMILY:mx8qm-nxp-bsp = "qm"
-SC_MX8_FAMILY:mx8qxp-nxp-bsp = "qx"
-SC_MACHINE_NAME = "mx8${SC_MX8_FAMILY}_b0"
+SC_MX8_FAMILY:var-som:mx8qm-nxp-bsp = "qm"
+SC_MX8_FAMILY:var-som:mx8qxp-nxp-bsp = "qx"
+SC_MACHINE_NAME:var-som = "mx8${SC_MX8_FAMILY}_b0"
 
-SCFW_BRANCH = "1.13.0"
-SRCREV = "052037315bdac7a5132a1bb1d475d6185def7789"
+SCFW_BRANCH:var-som = "1.13.0"
+SRCREV:var-som = "052037315bdac7a5132a1bb1d475d6185def7789"
 
-SRC_URI += " \
+SRC_URI:append:var-som = " \
     git://github.com/varigit/imx-sc-firmware.git;protocol=https;branch=${SCFW_BRANCH}; \
     https://developer.arm.com/-/media/Files/downloads/gnu-rm/8-2018q4/gcc-arm-none-eabi-8-2018-q4-major-linux.tar.bz2;name=gcc-arm-none-eabi \
 "
 
 SRC_URI[gcc-arm-none-eabi.md5sum] = "f55f90d483ddb3bcf4dae5882c2094cd"
 SRC_URI[gcc-arm-none-eabi.sha256sum] = "fb31fbdfe08406ece43eef5df623c0b2deb8b53e405e2c878300f7a1f303ee52"
 
-unset do_compile[noexec]
```

Comment:
Hi @kraj,
I am looking at finally integrating your changes into our langdale-var01 branch. One question, why did you make this change moving our compile functionality into `do_deploy()`?

Reply:
This makes it work in conjunction with latest meta-freescale and yocto, at yoe distro we use rolling release and always use master branches of all layers.

Reply:
Ok, but I think we should take a different approach rather than moving the task because our statements are compiling it. I see that leaving `unset do_compile[noexec]` would likely make your version run the default do_compile task (oe_runmake). What if instead we keep what you did in your earlier commit in this series e03c1143de30c07b9519aec323db32caa024e515 (correcting the MACHINEOVERRIDES typo)
```
do_compile:var-som() {
```
and then use a python anonymous function to dynamically delete the do_compile[noexec] flag in cases of var-som
```
python() {
    machine_overrides = d.getVar('MACHINEOVERRIDES').split(':')

    for override in machine_overrides:
        if override == "var-som":
            d.delVarFlag("do_compile", "noexec")
}
```
What do you think?